### PR TITLE
feat: use github app workflow

### PR DESCRIPTION
**Pull request by pradel**
Omg it worked, now the onboarding experience is super smooth 👀. It also means that private repo support is coming soon 🧙.

### CODE REVIEW

File: patches/smee-client+1.2.2.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/node_modules/smee-client/index.js b/node_modules/smee-client/index.js
+index 744cb0c..4e38100 100644
+--- a/node_modules/smee-client/index.js
++++ b/node_modules/smee-client/index.js
```
Comment (pradel) [inline patches/smee-client+1.2.2.patch:4]: Let's deploy this change on a fork to remove the need of patch-package

File: server/src/graphql/queries/setup.ts [hunk 2]
```diff
@@ -23,5 +23,34 @@ export const setup: QueryResolvers['setup'] = async () => {
...
+    // See https://docs.github.com/en/developers/apps/creating-a-github-app-from-a-manifest#examples
+    githubAppManifest: JSON.stringify({
+      name: 'Ledokku',
+      url:
+        process.env.NODE_ENV === 'production'
+          ? `http://${config.dokkuSshHost}`
+          : 'http://localhost:3000',
+      // TODO This should be enabled
+      // It's working for apps created via parameters but not here, not sure why
+      // request_oauth_on_install: true,
```
Comment (pradel) [inline server/src/graphql/queries/setup.ts:36]: probably the best is to ask in the Github support forum

File: server/src/graphql/mutations/loginWithGithub.ts [hunk 3]
```diff
@@ -80,6 +77,7 @@ export const loginWithGithub: MutationResolvers['loginWithGithub'] = async (
         avatarUrl: githubUser.avatar_url,
         email: email.email,
         githubAccessToken: data.access_token,
+        // TODO save refresh token + refresh_token_expires_in?
```
Comment (Akirtovskis) [inline server/src/graphql/mutations/loginWithGithub.ts:80]: @pradel should we create separate issue for this?

**Review by Akirtovskis — APPROVED** [review]
Good job @pradel , LGTM! 🚀🚀🚀

File: server/src/graphql/mutations/loginWithGithub.ts (hunk 3 above)
Comment (pradel) [reply server/src/graphql/mutations/loginWithGithub.ts:80]: @Akirtovskis yeah good idea to not forget about it 👍

**Pull request merged**